nx/lib/nx/random.ex

```diff
@@ -80,22 +80,18 @@ defmodule Nx.Random do
         [0, 12]
       >
 
-  A single key effectively consists of 64 bits, so all possible values
-  of a 64-bit integer result in a different key. However, when passing
-  an integer literal, Nx implicitly assumes its type to be `:s32`,
-  which would result in an overflow for large integers. Therefore,
-  when dealing with large seeds, make sure to explicitly use a 64 bit
-  type:
-
-      iex> Nx.Random.key(Nx.u64(999999999999))
+      iex> Nx.Random.key(999999999999)
       #Nx.Tensor<
         u32[2]
         [232, 3567587327]
       >
   """
-  defn key(seed) do
+  deftransform key(seed) do
     seed = Nx.as_type(seed, :u64)
```

Comment:
People may pass seed to defn and only then call `Nx.Random.key/1` (we do this in Bumblebee). But instead of documenting it, we can actually validate this:

```suggestion
    seed =
      case seed do
        seed when is_integer(seed) ->
          Nx.u64(seed)

        %Nx.Tensor{} = seed when seed.type == {:u, 64} ->
          seed
        
        other ->
          raise ArgumentError, "expected seed to be an integer or :u64 tensor, got: #{inspect(other)}"
      end

```

@polvalente @josevalim wdyt?

Reply:
I think this works, although it might make the function a bit less easy to use.

Reply:
@polvalente can you think of an example? I think most people would call it with an integer beforehand. If someone is passing seed to the defn, they probably use a default like `:erlang.system_time()`, which is a big number, so ideally we want to make sure they pass :u64.

Reply:
Not really. Maybe if the key call is deeply nested in the code, but that's more of an edge case I think

Reply:
@josevalim the thing is that the key has 64 bits, so if the user always crops the seed to 32 bits, multiple seeds may result in the same key, so it reduces the entropy.

Reply:
I'll commit this for the correctness. I think this would only actually raise in edge cases anyway.

Reply:
@jonatanklosko I was thinking specifically in the case of timestamps. The low level digits are always increasing while the higher ones are static, so we would be fine.

Reply:
Fair. But I would have the check anyway, if it turns out annoying, we can always disable and perhaps add some docs instead.